### PR TITLE
CI stability and Coveralls

### PR DESCRIPTION
**Pull request by josephpage**
- [x] Add *optional* `rbx-3` build to travis.
- [x] Make Travis builds more stable by adding `--retry 3` for `bundle install`
- [x] Update Coveralls integration with the [recommanded install way](https://docs.travis-ci.com/user/coveralls/)

### CODE REVIEW

**Comment by josephpage** [comment]
The failure is : 
```
Coverage (95.11%) is below the expected minimum coverage (95.33%).
```
Weird... the coverage with JRuby is smaller than with [MRI (95.85%)](https://travis-ci.org/oauth-xx/oauth2/jobs/326730470#L619).

**Comment by pboling** [comment]
*Need MOAR tests* 💃

**Comment by josephpage** [comment]
A Coveralls integration will be great to display and follow the detailled coverage ! 😉

**Comment by swrobel** [comment]
Any idea what's up w/ coveralls failing on Travis? https://travis-ci.org/oauth-xx/oauth2/jobs/326757613#L616

**Comment by pboling** [comment]
The tests are failing on a Rubocop error also.

**Comment by pboling** [comment]
It looks like this PR has changes we want to keep above what has already been applied to master. We should rebase and merge this.

File: README.md [hunk 1]
```diff
@@ -124,11 +124,12 @@ This library aims to support and is [tested against][travis] the following Ruby
...
 
 * Ruby 1.9.3
-* Ruby 2.0.0
+* Ruby 2.0
 * Ruby 2.1
 * Ruby 2.2
 * Ruby 2.3
 * Ruby 2.4
+* Ruby 2.5
 * [JRuby 9K][jruby]
```
Comment (pboling) [inline README.md]: We actually still support JRuby 1.7 in this release as well.  May as well list it too.

File: .travis.yml [hunk 2]
```diff
@@ -1,22 +1,18 @@
+language: ruby
+sudo: false
+
+env:
+  global:
+    - JRUBY_OPTS="-Xcli.debug=true --debug"
```
Comment (pboling) [inline .travis.yml]: The JRuby builds on this branch are failing hard, and I wonder if the change to JRUBY_OPTS is at fault?
```
JRUBY_OPTS="$JRUBY_OPTS --debug"
```
to
```
JRUBY_OPTS="-Xcli.debug=true --debug"
```

**Pull request merged**